paasta_tools/cli/cmds/local_run.py

```diff
@@ -617,32 +625,47 @@ def configure_and_run_docker_container(
     Run Docker container by image hash with args set in command line.
     Function prints the output of run command in stdout.
     """
-    try:
-        system_paasta_config = load_system_paasta_config()
-    except PaastaNotConfiguredError:
-        sys.stdout.write(PaastaColors.yellow(
-            "Warning: Couldn't load config files from '/etc/paasta'. This indicates\n"
-            "PaaSTA is not configured locally on this host, and local-run may not behave\n"
-            "the same way it would behave on a server configured for PaaSTA.\n"
-        ))
-        system_paasta_config = SystemPaastaConfig({"volumes": []}, '/etc/paasta')
 
     soa_dir = args.yelpsoa_config_root
 
     volumes = list()
 
-    instance_type = validate_service_instance(service, instance, cluster, soa_dir)
-
     load_deployments = docker_hash is None or pull_image
 
+    interactive = args.interactive
+
     try:
+        instance_type = validate_service_instance(service, instance, cluster, soa_dir)
         instance_config = get_instance_config(
             service=service,
             instance=instance,
             cluster=cluster,
             load_deployments=load_deployments,
             soa_dir=soa_dir,
         )
+    except NoConfigurationForServiceError as e:
+        if instance == 'interactive':
+            if not pull_image:
+                instance_type = 'adhoc'
+                instance_config = AdhocJobConfig(
+                    service=service,
+                    instance=instance,
+                    cluster=cluster,
+                    config_dict={
+                        'cpus': 1,
+                        'mem': 1024,
+                        'disk': 1024,
+                    },
+                    branch_dict={},
+                )
+                interactive = True
+            else:
+                sys.stderr.write(PaastaColors.red(
+                    "Error: you cannot use the default 'interactive' image with 'pull_image' specified.\n"))
```

Comment:
Ehhh crap. I guess I hadn't thought about how we would do this.

I guess we have to make a heuristic guesser  :(
Pick one of the deploy groups that any other instances are using that are defined in the same cluster.